src/Networking/NexusMods.Networking.Downloaders/Tasks/State/CompletedDownloadState.cs

```diff
@@ -0,0 +1,48 @@
+using NexusMods.Abstractions.MnemonicDB.Attributes;
+using NexusMods.MnemonicDB.Abstractions;
+
+namespace NexusMods.Networking.Downloaders.Tasks.State;
+
+/// <summary>
+/// Additional state for a <see cref="DownloaderState"/> that is completed
+/// </summary>
+public static class CompletedDownloadState
+{
+    private const string Namespace = "NexusMods.Networking.Downloaders.Tasks.DownloaderState";
+    
+    /// <summary>
+    /// The timestamp the download was completed at
+    /// </summary>
+    public static readonly DateTimeAttribute CompletedDateTime= new(Namespace, nameof(CompletedDateTime));
+    
+    /// <summary>
+    /// Whether the download is hidden (clear action) in the UI
+    /// </summary>
+    public static readonly BooleanAttribute Hidden = new(Namespace, nameof(Hidden));
+
+    /// <summary>
+    /// Model for reading and writing CompletedDownloadStates
+    /// </summary>
+    /// <param name="tx"></param>
+    public class Model(ITransaction tx) : DownloaderState.Model(tx)
+    {
+        
+        /// <summary>
+        /// The timestamp the download was completed at
+        /// </summary>
+        public DateTime CompletedAt
+        {
+            get => CompletedDateTime.Get(this, default(DateTime));
+            set => CompletedDateTime.Add(this, value);
+        }
```

Comment:
What's the reasoning behind not using the transaction timestamp?

Reply:
Since the entity in question is created at download start, we would need to use the timestamp of the transaction that sets the Status to `Completed`. 

This means that if the status changes to something else and then back to Completed, the timestamp would be wrong. 
For example I considered implementing Hidden as a new Status, which would have complicated this.
Other potential status could be QueryingMetadata or other things. Rather than trying to guess the correct timestamp, I thought this was just less complicated and future proof.